pkg/executor/build.go

```diff
@@ -370,6 +373,12 @@ func (s *stageBuilder) build() error {
 		}
 
 		if err := command.ExecuteCommand(&s.cf.Config, s.args); err != nil {
+			if s.opts.Debug {
+				logrus.Errorf(errors.Wrap(err, "Failed to execute command").Error())
+				if err := startDebugShell(s.opts.DebugCmd); err != nil {
+					return errors.Wrap(err, "error starting debug shell")
+				}
+			}
```

Comment:
I'm thinking about moving `startDebugShell()` under `command.ExecuteCommand()` and reuse `runCommandInExec()` which is actually loading a correct context with all the ENVs and ARGs set.

https://github.com/GoogleContainerTools/kaniko/blob/da50668f1eaffd16149719e275ba8d084b8d3e83/pkg/commands/run.go#L48-L50

But there is no options to pass command line arguments. Or I can modify `runCommandInExec()` to move all the common logic to separate function under `util` package.

What is the best option for this?